common/internal_model/src/test/scala/weco/catalogue/internal_model/work/StateTest.scala

```diff
@@ -0,0 +1,97 @@
+package weco.catalogue.internal_model.work
+import weco.catalogue.internal_model.work.WorkState.{
+  Denormalised,
+  Identified,
+  Merged
+}
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import weco.catalogue.internal_model.identifiers.IdentifierType.SierraIdentifier
+import weco.catalogue.internal_model.identifiers.{CanonicalId, SourceIdentifier}
+
+import java.time.Instant
+
+class StateTest
+    extends AnyFunSpec
+    with Matchers
+    with TableDrivenPropertyChecks {
```

Comment:
Couple of notes: we have a file WorkTest which has similar behaviour, and these tests probably belong in there. Good idea to add some tests here 👍

Also, there's a whole raft of traits called `{Something}Generators` that are quite useful for generating values in tests, usually when you don't care about the exact value.